huaweicloud/services/drs/data_source_huaweicloud_drs_availability_zones.go

```diff
@@ -0,0 +1,97 @@
+package drs
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API DRS POST /v3/{project_id}/available-zone
+func DataSourceAvailabilityZones() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceAvailabilityZonesRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"engine_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"direction": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"multi_write": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataSourceAvailabilityZonesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.DrsV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating DRS v3 client, error: %s", err)
+	}
+
+	listAZsHttpUrl := "v3/{project_id}/available-zone"
+	listAZsPath := client.Endpoint + listAZsHttpUrl
+	listAZsPath = strings.ReplaceAll(listAZsPath, "{project_id}", client.ProjectID)
+	listAZsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody: utils.RemoveNil(map[string]interface{}{
+			"engine_type":   d.Get("engine_type"),
+			"db_use_type":   d.Get("type"),
+			"job_direction": d.Get("direction"),
+			"node_type":     "high",
```

Comment:
it's better to export `node_type` as a parameter, so we can query the AZs with more flavors.